src/main/java/com/hms_networks/americas/sc/extensions/system/application/SCAppManagement.java

```diff
@@ -56,4 +61,72 @@ public static void disableAppAutoRestart() {
     Logger.LOG_DEBUG("Disabling next JVM run command.");
     RuntimeControl.configureNextRunCommand(jvmCommand);
   }
+
+  /**
+   * Checks if a valid WAN IP address has been assigned to the device.
+   *
+   * @return true if an initialized WAN IP address is found.
+   */
+  public static boolean deviceHasWanIP() {
+    final String uninitializedWanIp = "0.0.0.0";
+    return NetManager.getWanIp().equalsIgnoreCase(uninitializedWanIp);
+  }
+
+  /**
+   * Waits for a number of minutes for a WAN IP to be establlshed if one does not already exist.
+   * After the given time has passed, returns false indicating no WAN IP is present.
+   *
+   * @param timeOutMinutes returns true if an initialized WAN IP is detected
```

Comment:
incorrect javadoc

Reply:
I suggested correct javadocs in the above comment https://github.com/hms-networks/sc-ewon-flexy-extensions-lib/pull/22#discussion_r1002152290